rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -263,13 +318,18 @@ void serialize_array<std::string>(
     // First, cast field to rosidl_generator_c
     // Then convert to a std::string using StringHelper and serialize the std::string
     if (member->array_size_ && !member->is_upper_bound_) {
-        auto string_field = (rosidl_generator_c__String *) field;
-        ser.serializeArray(string_field->data, member->array_size_);
+      std::string stringseq[member->array_size_];
```

Comment:
fixed in https://github.com/eProsima/ROS-RMW-Fast-RTPS-cpp/pull/58/commits/8b844a55182d24e343573a87ff3bd0d0fb06b9e4